api/v1alpha1/timeout_types.go

```diff
@@ -45,6 +45,14 @@ type HTTPTimeout struct {
 	//
 	// +optional
 	RequestTimeout *gwapiv1.Duration `json:"requestTimeout,omitempty" yaml:"requestTimeout,omitempty"`
+
+	// StreamTimeout is the timeout for streaming requests. This timeout does not apply to non-streaming requests.
+	// When set to "0s", the timeout is disabled for streaming requests, allowing them to run indefinitely.
+	// This is particularly useful for gRPC streaming calls.
+	// Default: inherited from RequestTimeout.
```

Comment:
this doesnt look right
* request timeout is time diff b/w request sent and first response received (header counts)
* max stream timeout is time diff for stream completion (complete request sent and response received) and also applies to http1.1

Reply:
Stream timeout is specific to http2 right? What would you prefer to see here?

Reply:
it can also apply to SSE